docs/userguide/serialization.md

```diff
@@ -0,0 +1,384 @@
+mirroring the flexibility of specifying subtypes to your configuration file:
+
+```python
+from baybe.parameters.base import Parameter
+from baybe.parameters import CategoricalParameter, TaskParameter
+
+categorial_parameter = CategoricalParameter(name="setting", values=["low", "high"])
+categorical_parameter_str = """
+{
+    "type": "CategoricalParameter",
+    "name": "setting",
+    "values": ["low", "high"]
+}
+"""
+categorical_parameter_reconstructed = Parameter.from_json(categorical_parameter_str)
+assert categorial_parameter == categorical_parameter_reconstructed
+
+task_parameter = TaskParameter(name="setting", values=["low", "high"])
+task_parameter_str = """
+{
+    "type": "TaskParameter",
+    "name": "setting",
+    "values": ["low", "high"]
+}
+"""
+task_parameter_reconstructed = Parameter.from_json(task_parameter_str)
+assert task_parameter == task_parameter_reconstructed
+```
+
+```{note} 
+When serializing an object that belongs to a class hierarchy, BayBE automatically
+injects the `type` field into the serialization string to enable frictionless deserialization
+at a later stage.
+```
+
+### Using abbreviations
+Classes that have an `_abbreviation` class variable defined can be conveniently
+deserialization using the corresponding abbreviation string:
+```python
+from baybe.acquisition.base import AcquisitionFunction
+
+acqf1 = AcquisitionFunction.from_json('{"type": "UpperConfidenceBound"}')
+acqf2 = AcquisitionFunction.from_json('{"type": "UCB"}')
+
+assert acqf1 == acqf2
+```
+
+### Nesting objects
+BayBE objects typically appear as part of a larger object hierarchy.
+For instance, a
+{class}`SearchSpace <baybe.searchspace.core.SearchSpace>` can hold one or several
+{class}`Parameters <baybe.parameters.base.Parameter>`, just like an
+{class}`Objective <baybe.objectives.base.Objective>` can hold one ore several 
+{class}`Targets <baybe.targets.base.Target>`.
+This hierarchical structure can be directly replicated in the serialization string:
+
+```python
+from baybe.objectives import DesirabilityObjective
+from baybe.targets import NumericalTarget
+
+objective = DesirabilityObjective(
+    targets=[
+        NumericalTarget(name="t1", mode="MAX", bounds=(-1, 1)),
+        NumericalTarget(name="t2", mode="MIN", bounds=(0, 1)),
+    ],
+    weights=[0.1, 0.9],
+    scalarizer="MEAN",
+)
+
+objective_str = """
+{   
+    "type": "DesirabilityObjective", 
+    "targets": [
+        {
+            "type": "NumericalTarget",
+            "name": "t1",
+            "mode": "MAX",
+            "bounds": [-1.0, 1.0]
+        }, 
+        {
+            "type": "NumericalTarget",
+            "name": "t2",
+            "mode": "MIN",
+            "bounds": [0.0, 1.0]
+        }
+    ],
+    "weights": [0.1, 0.9],
+    "scalarizer": "MEAN"
+}
+"""
+
+assert objective == DesirabilityObjective.from_json(objective_str)
+```
+
+(ALTERNATIVE_CONSTRUCTORS)=
+### Invoking alternative constructors
```

Comment:
Yes, but I think search spaces are the primary example for this. I quickly searched but don't think we really have any other useful example at the moment (apart from dataframes, for which there is a separate section, and Intervals, which are rather uninteresting and whose constructors are not suited for the case). 

If you have another idea, please let me know. Otherwise, please resolve (and perhaps we can change it it the future when there is another case)

Reply:
ok thats fine then, the example is good indeed